contentcuration/contentcuration/frontend/channelEdit/views/ImportFromChannels/BrowsingCard.vue

```diff
@@ -192,30 +181,48 @@
         return get(this.node, ['description', 'length']) > 120;
       },
       numLocations() {
-        return 1;
+        return this.node.location_ids.length;
       },
       goToLocationLabel() {
         return this.numLocations > 1
           ? this.$tr('goToPluralLocationsAction', { count: this.numLocations })
           : this.$tr('goToSingleLocationAction');
       },
-      goToLocationRoute() {
-        return {
-          name: this.RouterNames.IMPORT_FROM_CHANNELS_BROWSE,
-          params: {
-            channelId: this.node.source_channel_id,
-            nodeId: this.node.parent,
-          },
-          query: {
-            nodeId: this.node.id,
-          },
-        };
+      isTopic() {
+        return this.node.kind === 'topic';
       },
     },
     methods: {
-      emitPreview() {
-        this.$emit('preview');
+      ...mapActions('clipboard', ['copy']),
+      handleClick() {
```

Comment:
Similarly to the store dependency, I would welcome to not have router dependency in here unless necessary. In addition to that, having two handlers if click event in different components, one of them being triggered only under certain conditions that are defined only in the component itself, seems to be a bit confusing.

Reply:
Sounds good! I think I'm going to address this later since there are other router dependencies on the component and may need more planning